### ISSUE TITLE
[Import API] Error in CSV delimiter detection

### ISSUE THREAD

**Issue by manmorjim** [CartoDB/cartodb#15423]
When we are importing a CSV with, for example, _comma_ separator but with a huge number of spaces the [CSV Normalizer](https://github.com/CartoDB/cartodb/blob/master/services/importer/lib/importer/csv_normalizer.rb) is [guessing](https://github.com/CartoDB/cartodb/blob/123d5affdf4c1712f0825259da537de521a06518/services/importer/lib/importer/csv_normalizer.rb#L60) _space_ as delimiter of the file which is wrong.

We should improve the delimiter detection by [giving more importance/weight to _comma_ delimiter](https://github.com/CartoDB/cartodb/blob/123d5affdf4c1712f0825259da537de521a06518/services/importer/lib/importer/csv_normalizer.rb#L17) due to it is the most common separator used in CSV.

**Comment by Jesus89** [comment]
Hi. I tried comma-separated CSVs with lots of spaces and I did not reproduce the issue.

I tested also the CSV file provided by @manmorjim and I had the same behavior, but in this case, I think there is an issue with the CSV file itself because it does not contain row delimiters either headers.

**Comment by Jesus89** [comment]
I have reproduced the issue of delimiter detection. The problem is not with the weights, because they are only used when there is one line. For multiple lines, it gets the variance of the occurrence of the candidates.

In the first CSV case the variances are:
- ","=>2624.9392929292935
- " "=>633.0036363636376
So the selected delimiter is the one with less variance.

I have reduced the LINES_FOR_DETECTION to 20 and it detects the "," correctly, but this is not a good solution.

In python, the csv module has a method to detect the delimiter, and this works OK with the two provided CSV files that fail in the CARTO import. This is the function: https://github.com/python/cpython/blob/b10dc3e7a11fcdb97e285882eba6da92594f90f9/Lib/csv.py#L280.

I'll check that algorithm to see if we can adjust the `detect_delimiter` function in the import.
